src/v/raft/recovery_stm.cc

```diff
@@ -33,7 +35,23 @@ recovery_stm::recovery_stm(
   , _scheduling(scheduling)
   , _ctxlog(_ptr->_ctxlog) {}
 
-ss::future<> recovery_stm::do_recover() {
+ss::future<> recovery_stm::recover() {
+    auto meta = get_follower_meta();
+    if (!meta) {
+        // stop recovery when node was removed
+        _stop_requested = true;
+        return ss::now();
+    }
```

Comment:
Is this a separate commit, comment, or addition to the commit message? It seems orthogonal to applying the scheduling group and iopc and its not clear what case its handling which wasn't present prior to this change.

Reply:
this check is part of applying scheduling group. it handled in different places. If node was removed from the raft group we have to stop recovery. We use follower state (i.e. `meta` variable) to determine which scheduling group to apply.

Reply:
oh I see. why duplicate the check if it is the first thing that do_recover does?

Reply:
it is already after  a scheduling point few lines bellow, recovery stm works outside of raft lock

Reply:
meta is used to check the learner state to affect the scheduling group used, so we need to repeat the same check not because of any races or synchronization between foo and do_foo but because meta may simply not be set.